packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -298,21 +187,21 @@ library ExchangeBlocks
                 validateAndUpdateProtocolFeeValues(S, protocolTakerFeeBips, protocolMakerFeeBips),
                 "INVALID_PROTOCOL_FEES"
             );
-        } else if (blockType == ExchangeData.BlockType.DEPOSIT) {
+        } else if (_block.blockType == ExchangeData.BlockType.DEPOSIT) {
             uint startIdx = 0;
             uint count = 0;
             assembly {
                 startIdx := and(mload(add(data, 136)), 0xFFFFFFFF)
                 count := and(mload(add(data, 140)), 0xFFFFFFFF)
             }
-            require (startIdx == numDepositRequestsCommitted, "INVALID_REQUEST_RANGE");
-            require (count <= blockSize, "INVALID_REQUEST_RANGE");
-            require (startIdx + count <= S.depositChain.length, "INVALID_REQUEST_RANGE");
+            require (startIdx == numDepositRequestsCommitted, "INVALID_REQUEST_RANGE_1");
+            require (count <= _block.blockSize, "INVALID_REQUEST_RANGE_2");
+            require (startIdx + count <= S.depositChain.length, "INVALID_REQUEST_RANGE_3");
 
             bytes32 startingHash = S.depositChain[startIdx - 1].accumulatedHash;
             bytes32 endingHash = S.depositChain[startIdx + count - 1].accumulatedHash;
             // Pad the block so it's full
-            for (uint i = count; i < blockSize; i++) {
+            for (uint i = count; i < _block.blockSize; i++) {
```

Comment:
`sha256` is exactly the same as `fastSHA256`. `fastSHA256` is just a version I made that is more efficient than the solidity implementation for large amounts of data because it does less memory copies.